### ISSUE TITLE
Does puppeteer-extra use js to replace browser settings?

### ISSUE THREAD

**Issue by qo4on** [berstend/puppeteer-extra#207]
If so stealth is no longer a stealth. Read this [article](https://cpa.rip/stati/antidetect-palivo/) (Google translate to English).

**Comment by StevenVeshkini** [comment]
Yeah, seems like Headless Chrome can be detected quite easily with the techniques in that article...For example, running:

`if( Object.getOwnPropertyNames(navigator)[0] ) alert('fake parameters detected');`

after loading a page in the stealth plugin will cause the alert, but wont in regular Chrome...

**Comment by qo4on** [comment]
It's very intresting to know what @berstend and @Niek think about it.

**Comment by Niek** [comment]
That's easy to work around: always copy the prototype of an object, make changes and then copy it back.
```javascript
const newProto = navigator.__proto__;
Object.defineProperty(newProto, 'languages', {get: () => ['my', 'non-existing', 'languages']});
navigator.__proto__ = newProto;
navigator.languages
     (3) ["my", "non-existing", "languages"]
Object.getOwnPropertyNames(navigator)
     []
```

**Comment by qo4on** [comment]
Puppeteer stealth does not work. Try to create a gmail account and you get an error.

**Comment by berstend** [comment]
Fixed in #275, #276, #277, #278 😄
